app/pages/clinicworkspace/ClinicPatients.js

```diff
@@ -2417,21 +2427,56 @@ export const ClinicPatients = (props) => {
               )}
             </Formik>
 
-            <Text mb={2} sx={{ color: 'text.primary', fontWeight: 'medium', fontSize: 0 }}>
-              {isClinicAdmin
-                ? t('Click a tag\'s text to rename it, or click the trash can icon to delete it.')
-                : t('Click a tag\'s text to rename it.')
+            { patientTagsFilterOptions.length > 0 &&
+              <>
+                <Box>
+                  <Text sx={{ fontSize: 1, color: 'text.primary', fontWeight: 'medium' }}>
+                    {t('Tags ({{ count }})', { count: clinic?.patientTags?.length || '0' })}{' - '}
+                  </Text>
+                  <Text sx={{ fontSize: 0, color: 'text.primary' }}>
+                    {t('Click on the edit icon to rename the tag or trash icon to delete it.')}
+                  </Text>
+                </Box>
+                <Box mt={1} mb={0}>
+                  <Text sx={{ fontSize: 0, color: colors.gray50, fontStyle: 'italic' }}>
+                    {t('Name')}
+                  </Text>
+                </Box>
+              </>
+            }
+
+            <Box mt={1} id="clinic-patients-edit-tag-list">
+              {
+                clinic?.patientTags?.map(({ id, name }) => (
+                  <Grid py={2} sx={{
+                    gridTemplateColumns: '1fr 72px 16px',
+                    borderTop: `1px solid ${colors.gray05}`,
+                    alignItems: 'center',
+                  }}>
+                    <Flex sx={{ alignItems: 'center'}}>
+                      <Text className="tag-text" sx={{ fontSize: 1, color: 'text.primary' }}>{name}</Text>
+                      <Icon
+                        id={`edit-tag-button-${id}`}
+                        icon={EditIcon}
+                        sx={{ fontSize: 1, marginLeft: 2 }}
+                        onClick={isClinicAdmin ? () => handleUpdateClinicPatientTag(id) : undefined}
+                      />
+                    </Flex>
+                    <Box>
+                      {/* TODO: Insert per-tag patient counts here */}
```

Comment:
Just calling out to check ig this is intended to be left in a TODO state for this PR.  Not blocking approval over it.